### PR TITLE
refactor(material): 物料管理插件元应用拆分

### PR DESCRIPTION
**Pull request by wenmine**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Added metadata entries for different components (header, content, block, component) to enhance plugin configuration.
  - Introduced new components: `MaterialLayout`, `MaterialBlock`, `MaterialComponent`, and `MaterialHeader`.

- **Improvements**
  - Updated import paths for better organization and code clarity.
  - Enhanced dynamic rendering of components based on layout and metadata.
  - Simplified import statements to improve readability.

- **Bug Fixes**
  - Fixed various import statement inconsistencies.
  - Corrected trailing comma in `BreadcrumbService` declaration.

- **Style**
  - Updated import statements to use single quotes for consistency.

### CODE REVIEW

**Comment by wenmine** [comment]
image

**Review by hexqi — DISMISSED** [review]

File: packages/plugins/materials/src/meta/layout/src/Main.vue [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      emitEvent: emit
+    })
+    provide('panelState', panelState) // 使用provide传给子组件,后续可能会有调整，先暂定
+
+    const pluginRegistryData = ref(props.registryData)
+    const rightPanelRef = ref(null)
+
+    const headerComponent = getMergeMeta(pluginRegistryData.value.components?.header)
+    const onlyShowDefault = ref(pluginRegistryData.value.options.childrenIds?.length === 1)
+    const activeName = ref(pluginRegistryData.value.options?.defaultTabId)
```
Comment (hexqi) [inline packages/plugins/materials/src/meta/layout/src/Main.vue]: activeName没有配置时可以默认选中第一个

File: packages/plugins/materials/src/meta/layout/src/Main.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    const defaultComponent = getMergeMeta(activeName.value)?.component
```
Comment (hexqi) [inline packages/plugins/materials/src/meta/layout/src/Main.vue]: 这里应该是entry

File: packages/plugins/materials/src/meta/layout/src/Main.vue [hunk 3]
```diff
@@ -50,13 +50,16 @@ export default {
 
     const pluginRegistryData = ref(props.registryData)
     const rightPanelRef = ref(null)
-
+    const tabIds = pluginRegistryData.value.options.tabIds || []
```
Comment (hexqi) [inline packages/plugins/materials/src/meta/layout/src/Main.vue]: tabIds为空数组看起来没有意义，界面上面都没有，建议改为没有配置时显示所有

File: packages/plugins/materials/src/meta/layout/src/Main.vue [hunk 4]
```diff
@@ -50,13 +50,16 @@ export default {
...
     const headerComponent = getMergeMeta(pluginRegistryData.value.components?.header)
-    const onlyShowDefault = ref(pluginRegistryData.value.options.childrenIds?.length === 1)
-    const activeName = ref(pluginRegistryData.value.options?.defaultTabId)
+    const onlyShowDefault = ref(tabIds.length === 1)
+    const activeTabId =
+      pluginRegistryData.value.options.tabIds.find((item) => item === pluginRegistryData.value.options?.defaultTabId) ||
```
Comment (hexqi) [inline packages/plugins/materials/src/meta/layout/src/Main.vue]: 没有配置tabIds时find方法会报错。上面已经定义了tabIds变量了，这里可以直接用上面的变量

File: packages/plugins/materials/index.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ export default {
   layout: MaterialLayout,
   options: {
     defaultTabId: 'engine.plugins.materials.component',
-    childrenIds: ['engine.plugins.materials.component', 'engine.plugins.materials.block']
+    tabIds: ['engine.plugins.materials.component', 'engine.plugins.materials.block']
```
Comment (hexqi) [inline packages/plugins/materials/index.js]: tabIds看起来是用来配置要显示的tab页签的，感觉可以再换个名字

Comment (wenmine) [reply packages/plugins/materials/index.js]: 改成tabComponentIds，看看合适吗

File: packages/plugins/materials/src/meta/layout/src/Main.vue (hunk 3 above)
Comment (wenmine) [reply packages/plugins/materials/src/meta/layout/src/Main.vue]: 显示所有是显示哪里的所有，是metas字段吗，但是现在metas里也有元服务，需要对metas里的元应用进行过滤获取嘛？感觉很麻烦，这边没有配置其实就是默认用我的缺省配置了，如果配置了就是要调整的意思吧？

**Review by hexqi — APPROVED** [review]

File: packages/plugins/materials/index.js (hunk 5 above)
Comment (hexqi) [reply packages/plugins/materials/index.js]: 我没有说明白，我的意思其实是说，名字体现出配置的是要显示的,例如，visibleTabIds 或 displayedTabIds，你看下这样会不会好一点

**Pull request merged**